### PR TITLE
Fixes for issues found during Github Project setup

### PR DESCRIPTION
**Pull request by IvanSavenko**
@DjWarmonger @misiokles 
Can you check this?

While setting up HotA status project found several mismatches/missing parameters.
Fixed those that were straightforward.

Some are still left unsolved/unchecked, mostly specialties - not 100% sure how those updaters work

Also added Ukrainian translation since I was planning it anyway.

### CODE REVIEW

File: hota/Mods/HotA banks/Content/config/banks/banks/pirateCavern.json [hunk 1]
```diff
@@ -20,19 +20,18 @@
...
 				[
 					{
 						"chance": 30,
 						"guards": [
-							{ "amount": 10, "type" : "pirate" },
-							{ "amount": 10, "type" : "pirate" },
-							{ "amount": 10, "type" : "corsair", "upgradeChance": 50 },
-							{ "amount": 10, "type" : "pirate" },
-							{ "amount": 10, "type" : "corsair" }
+							{ "amount": 20, "type" : "pirate", "upgradeChance": 50 },
```
Comment (misiokles) [inline hota/Mods/HotA banks/Content/config/banks/banks/pirateCavern.json]: According to: https://heroes.thelazy.net//index.php/Pirate_Cavern old guards are more correct. In this variant there are 40 guards dividen into 4 stacks.

File: hota/Mods/HotA banks/Content/config/banks/banks/pirateCavern.json [hunk 2]
```diff
@@ -20,19 +20,18 @@
...
-					"value"		: 2500,
+					"value"		: 3500,
 					"rarity"	: 100
 				},
 				"levels":
 				[
 					{
 						"chance": 30,
 						"guards": [
-							{ "amount": 10, "type" : "pirate" },
```
Comment (misiokles) [inline hota/Mods/HotA banks/Content/config/banks/banks/pirateCavern.json:31]:
```
							{ "amount": 10, "type" : "pirate", "upgradeChance": 50 },
							{ "amount": 10, "type" : "pirate" },
							{ "amount": 10, "type" : "corsair" },
							{ "amount": 10, "type" : "corsair"}
```

File: hota/Mods/HotA banks/Content/config/banks/banks/spit.json [hunk 3]
```diff
@@ -13,35 +13,32 @@
 						"animation" : "banks/map/banks/avbone02.def", 
 						"visitableFrom" : [ "---", "-+-", "+++" ], 
 						"mask" : [ "VVVV", "VVBB", "VBAB"],
-						"allowedTerrains":["grass", "swamp"] 							
+						"allowedTerrains":["grass", "swamp"]
```
Comment (misiokles) [inline hota/Mods/HotA banks/Content/config/banks/banks/spit.json]: According to wiki: https://heroes.thelazy.net//index.php/Spit only in swamp terrain should be placed. And stack placement is diffent, but I guess it's not implemented in vcmi.

File: hota/Mods/HotA banks/Content/config/banks/banks/pirateCavern.json [hunk 4]
```diff
@@ -20,19 +20,18 @@
...
-							{ "amount": 10, "type" : "pirate" },
-							{ "amount": 10, "type" : "corsair", "upgradeChance": 50 },
-							{ "amount": 10, "type" : "pirate" },
-							{ "amount": 10, "type" : "corsair" }
+							{ "amount": 20, "type" : "pirate", "upgradeChance": 50 },
+							{ "amount": 20, "type" : "corsair" },
+							{ "amount": 20, "type" : "corsair" },
+							{ "amount": 20, "type" : "pirate" },
```
Comment (misiokles) [inline hota/Mods/HotA banks/Content/config/banks/banks/pirateCavern.json]: Remove comma.

File: hota/Mods/HotA banks/Content/config/banks/banks/pirateCavern.json [hunk 5]
```diff
@@ -43,11 +42,10 @@
 					{
 						"chance": 30,
 						"guards": [
-							{ "amount": 20, "type" : "pirate" },
-							{ "amount": 20, "type" : "pirate" },
-							{ "amount": 20, "type" : "corsair", "upgradeChance": 50 },
-							{ "amount": 20, "type" : "pirate" },
-							{ "amount": 20, "type" : "corsair" }
+							{ "amount": 40, "type" : "pirate", "upgradeChance": 50 },
```
Comment (misiokles) [inline hota/Mods/HotA banks/Content/config/banks/banks/pirateCavern.json]: Should be 2-x4 stacks

File: hota/Mods/HotA banks/Content/config/banks/banks/pirateCavern.json [hunk 6]
```diff
@@ -43,11 +42,10 @@
...
+							{ "amount": 40, "type" : "corsair" },
+							{ "amount": 40, "type" : "corsair" },
+							{ "amount": 40, "type" : "pirate" },
```
Comment (misiokles) [inline hota/Mods/HotA banks/Content/config/banks/banks/pirateCavern.json]: Remove comma.

File: hota/Mods/HotA banks/Content/config/banks/banks/pirateCavern.json [hunk 7]
```diff
@@ -58,11 +56,10 @@
 					{
 						"chance": 30,
 						"guards": [
-							{ "amount": 30, "type" : "pirate" },
-							{ "amount": 30, "type" : "pirate" },
-							{ "amount": 30, "type" : "corsair", "upgradeChance": 50 },
-							{ "amount": 30, "type" : "pirate" },
-							{ "amount": 30, "type" : "corsair" }
+							{ "amount": 60, "type" : "pirate", "upgradeChance": 50 },
```
Comment (misiokles) [inline hota/Mods/HotA banks/Content/config/banks/banks/pirateCavern.json]: Should be 30 x 4 stacks

File: hota/Mods/HotA banks/Content/config/banks/banks/pirateCavern.json (hunk 1 above)
Comment (IvanSavenko) [reply hota/Mods/HotA banks/Content/config/banks/banks/pirateCavern.json]: Fixed. Now bank has 4 stacks with 10/20/30/40 guards, which matches HotA

File: hota/Mods/HotA banks/Content/config/banks/banks/spit.json (hunk 3 above)
Comment (IvanSavenko) [reply hota/Mods/HotA banks/Content/config/banks/banks/spit.json]: Stack placement - yeah, trivial to add though.
I've made separate item about that in VCMI-HotA project.
Same issue with churchyard bank.

**Comment by misiokles** [comment]
About Hota hero specialties:
- I'm not sure about Zilare if really his specialty makes forgetfulness last longer.
- Eovacius: it's impossible to implement his specialty 1:1, but Andruids made something cool instead and gave his special clone spell.
All others are fine.

**Comment by IvanSavenko** [comment]
What I've found with specialties:
- Beatrice & Miriam: bonus (presumably) matches HotA, but description is incorrect:
Should be: "Increases the scouting range by 1 for every 6 levels of the hero"
Currently: "Receives a 5% per level bonus to Scouting skill percentage (rounded up)."

- Kinkeria: different from HotA:
HotA: Learning skill percentage is doubled.
VCMI: Receives a 5% per level bonus to Learning skill percentage.

- Not a bug, but all creature specialists can use "short" form that only specifies target creature (except for Sea Dogs specialist, fixed for vcmi-1.3)

- Leena: specialty is OK, but starting skill is missing - should also have pathfinding. Missed that one.

**Pull request merged**